### DISCUSSION TITLE
Errors in compiling code

### DISCUSSION THREAD

**Discussion by nikandriko** [phord/Jarvis#20] [Q&A]
@phord I'm getting this error when I verify the code on the Arduino IDE.

I suspect I might be missing some library.

`c:/users/xx/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.0.4-gcc10.3-1757bed/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: sketch\Jarvis.ino.cpp.o: in function _ZN12TelnetLogger9print_rawIPKcJ6StringcEEEvRKT_DpT0_:`

`C:\Users\xx\Documents\Arduino\5. Debugging\Jarvis\src/TelnetLogger.h:36: undefined reference to _ZN3Ota5beginEv`

`c:/users/xx/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.0.4-gcc10.3-1757bed/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: sketch\Jarvis.ino.cpp.o: in function setup:`

`C:\Users\xx\Documents\Arduino\5. Debugging\Jarvis/Jarvis.ino:71: undefined reference to _ZN3Ota5beginEv`

`c:/users/xx/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.0.4-gcc10.3-1757bed/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: C:\Users\xx\Documents\Arduino\5. Debugging\Jarvis/Jarvis.ino:34: undefined reference to _ZN3Ota4loopEv`

`c:/users/xx/appdata/local/arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.0.4-gcc10.3-1757bed/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: sketch\Jarvis.ino.cpp.o: in function loop:`

`C:\Users\xx\Documents\Arduino\5. Debugging\Jarvis/Jarvis.ino:39: undefined reference to _ZN3Ota4loopEv`

`collect2.exe: error: ld returned 1 exit status`

`exit status 1`

`Error compiling for board LOLIN(WEMOS) D1 R2 & mini.`

**Comment by phord** [comment]
I haven't seen those before.  It does look like it's missing a library, or that the function signatures are wrong for it.

Are you able to build 'blink.ino' for your Wemos successfully?

**Comment by imightbelosthere** [comment]
Check your ota.cpp file. There are 2 with the same name and Windows overwrites the one with code by the one that's empty.
